ci/docker/conda-python-substrait.dockerfile

```diff
@@ -0,0 +1,45 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ARG repo
+ARG arch
+ARG python=3.9
+
+FROM ${repo}:${arch}-conda-python-${python}
+
+COPY ci/conda_env_python.txt \
+     ci/conda_env_sphinx.txt \
+     /arrow/ci/
+RUN mamba install -q -y \
+        --file arrow/ci/conda_env_python.txt \
+        --file arrow/ci/conda_env_sphinx.txt \
+        $([ "$python" == "3.9" ] && echo "pickle5") \
+        python=${python} openjdk \
+        nomkl && \
+    mamba clean --all
+
+
+ARG substrait=latest
+COPY ci/scripts/install_substrait_consumer.sh /arrow/ci/scripts/
+
```

Comment:
could we run the `install_substrait_consumer.sh` here instead of on the command? Similar to what we do on the spark installation for the spark integration tests: https://github.com/apache/arrow/blob/master/ci/docker/conda-python-spark.dockerfile#L35
That way the installation is cached on the image and we don't have to run them every time we run a container.
If we can't, the only reason I can think of is we require to build arrow/pyarrow first, we probably can reuse the base [conda-python.dockerfile](https://github.com/apache/arrow/blob/master/ci/docker/conda-python.dockerfile) and just execute the `install_substrait_consumer.sh` which should already be available on the container without requiring the explicit copy instead of creating a new dockerfile.

Reply:
@raulcd I see your point. I have updated the script, but first let me test it and ping you.

Reply:
@raulcd I think I got stuck where not having pyarrow installed in the system. Please note that there is another dependency `openjdk` installed when mamba creates the environment. By the way how can we re-use the `conda-python.dockerfile` in this case? I don't know much about CIs in depth.

Reply:
Thanks @vibhatha ! 
I've tested locally and added some comments. We can install substrait consumer on build but running the integration tests should be done as part of the command as, as you suggested, requires pyarrow to be built.
As per the new dependency for `openjdk` I missed that, sounds good to me!